src/paulscode/android/mupen64plusae/profile/TouchscreenProfileActivity.java

```diff
@@ -385,37 +397,120 @@ private void hideSystemBars()
     @Override
     public boolean onTouch( View v, MotionEvent event )
     {
-        if( ( event.getAction() & MotionEvent.ACTION_MASK ) != MotionEvent.ACTION_DOWN )
-            return false;
-        
         int x = (int) event.getX();
         int y = (int) event.getY();
         
-        // Get the N64 index of the button that was pressed
-        int index = mTouchscreenMap.getButtonPress( x, y );
-        if( index != TouchMap.UNMAPPED )
+        if( ( event.getAction() & MotionEvent.ACTION_MASK ) == MotionEvent.ACTION_DOWN )
         {
-            String assetName = TouchMap.ASSET_NAMES.get( index );
-            String title = READABLE_NAMES.get( index );
+            initialX = x;
+            initialY = y;
+            dragIndex = TouchMap.UNMAPPED;
+            dragging = false;
             
-            // D-pad buttons are not holdable
-            if( DPAD.equals( assetName ) )
-                index = -1;
+            // ignore touches that start at the very bottom or top of the screen
+            View view = getWindow().getDecorView();
+            if (y < 10 || y > view.getHeight() - 10)
+                return false;
             
-            popupDialog( assetName, title, index );
+            // Get the N64 index of the button that was pressed
+            int index = mTouchscreenMap.getButtonPress( x, y );
+            if( index != TouchMap.UNMAPPED )
+            {
+                dragIndex = index;
+                dragAsset = TouchMap.ASSET_NAMES.get( index );
+                dragFrame = mTouchscreenMap.getButtonFrame( dragAsset );
+            }
+            else
+            {
+                // See if analog was pressed
+                Point point = mTouchscreenMap.getAnalogDisplacement( x, y );
+                int dX = point.x;
+                int dY = point.y;
+                float displacement = FloatMath.sqrt( ( dX * dX ) + ( dY * dY ) );
+                if( mTouchscreenMap.isInCaptureRange( displacement ) )
+                {
+                    dragIndex = -2;
```

Comment:
Should we use a constant for this? I just chose a number that wasn't being used, since the analog stick doesn't seem to have its own index value.